// vso-node-api/handlers/apiversion.js
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.
"use strict";
var ApiVersionHandler = (function () {
    function ApiVersionHandler(apiVersion) {
        this.apiVersion = apiVersion;
    }
    // currently implements pre-authorization
    // TODO: support preAuth = false where it hooks on 401
    ApiVersionHandler.prototype.prepareRequest = function (options) {
        if (options.headers && options.headers['Accept']) {
            options.headers["Accept"] = options.headers['Accept'] + '; api-version=' + this.apiVersion;
        }
    };
    // This handler cannot handle 401
    ApiVersionHandler.prototype.canHandleAuthentication = function (res) {
        return false;
    };
    ApiVersionHandler.prototype.handleAuthentication = function (httpClient, protocol, options, objs, finalCallback) {
    };
    return ApiVersionHandler;
}());
exports.ApiVersionHandler = ApiVersionHandler;
